memory-viz/src/memory_model.ts

```diff
@@ -1145,4 +1204,96 @@ export class MemoryModel {
     }
+
+    /**
+     * Add hover interactivity to the SVG on object IDs
+     */
+    setInteractivityScript(): void {
+        const idToObjectMapping = Object.fromEntries(this.idToObjectMap);
+
+        const script = `
+            function objectInteractivity() {
+                // Inject the id value to object id mapping into script
+                const idToObjectMap = ${JSON.stringify(idToObjectMapping)};
+                
+                function highlightObject(objectId) {
+                    const objectBox = document.getElementById(objectId);
+                    if (!objectBox) return;
+                    
+                    const bbox = objectBox.getBBox();
+                    const existingFill = objectBox.querySelector('path[fill]:not([fill="none"])');
+                    
+                    if (existingFill) {
+                        existingFill.setAttribute('data-original-fill', existingFill.getAttribute('fill'));
+                        existingFill.setAttribute('fill', 'rgba(255, 255, 0, 0.6)');
+                    } else {
+                        const highlight = document.createElementNS("http://www.w3.org/2000/svg", "rect");
+                        highlight.setAttribute("x", bbox.x);
+                        highlight.setAttribute("y", bbox.y);
+                        highlight.setAttribute("width", bbox.width);
+                        highlight.setAttribute("height", bbox.height);
+                        highlight.setAttribute("fill", "rgba(255, 255, 0, 0.6)");
+                        highlight.setAttribute("pointer-events", "none");
+                        highlight.setAttribute("id", objectId + "-highlight");
+                        
+                        objectBox.insertBefore(highlight, objectBox.firstChild);
+                    }
+                }
+                
+                function removeHighlight(objectId) {
+                    const objectBox = document.getElementById(objectId);
+                    if (!objectBox) return;
+                    
+                    const existingFill = objectBox.querySelector('path[data-original-fill]');
+                    
+                    if (existingFill) {
+                        existingFill.setAttribute('fill', existingFill.getAttribute('data-original-fill'));
+                        existingFill.removeAttribute('data-original-fill');
+                    } else {
+                        const highlight = document.getElementById(objectId + "-highlight");
+                        if (highlight) highlight.remove();
+                    }
+                }
+                
+                function addEventListeners() {
+                    document.querySelectorAll('text.id').forEach(idText => {
+                        const idValue = idText.textContent.trim();
+                        if (!idValue || !idValue.startsWith('id')) return;
+                        
+                        idText.style.cursor = 'pointer';
```

Comment:
This should really be defined in the CSS for all `text.id` elements (but only when `interactive` is `true`)